ExcelWriter.java

```diff
@@ -0,0 +1,69 @@
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.*;
+import javax.swing.JFileChooser;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFCell;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+public class ExcelWriter extends ReDoReport {
+	static JFileChooser fileChooser;
+	 public static final String OUTPUT_FILE = "Hours_" + FileUtility.getTimeStamp() +
+				".xlsx";
+	
+	 public static void outputFile(List<String> projects, List<Date> dates,
+			List<String> names, List<Double> duration) throws IOException{
+		String[] header = {"Month", "Customer", "Project", "Hours", "Cost", 
+				"Invoiced", "Profitability"};
+		String[] dateHeader = {" ","Jan","Feb","Mar","Apr", "May","Jun","Jul",
+				"Aug","Sep","Oct","Nov","Dec"};
+		Calendar now = Calendar.getInstance();
+		int year = now.get(Calendar.YEAR);
+		XSSFWorkbook wb = new XSSFWorkbook();
+		XSSFSheet sheet0 = wb.createSheet("Profitability");
+		XSSFSheet sheet1 = wb.createSheet("Utilization");
+		XSSFSheet sheet2 = wb.createSheet("Original File");
+		CellStyle headCellStyle = wb.createCellStyle();
+		Row pHeadRow = sheet0.createRow(0);
+		Row client = sheet0.createRow(1);
+		Row uHeadRow = sheet1.createRow(0);
+		Row initHeadRow = sheet2.createRow(0);
+		
+		for (int i = 0; i < header.length; i++) {
+			Cell cell = pHeadRow.createCell(i);
+			cell.setCellValue(header[i]);
+			cell.setCellStyle(headCellStyle);
+		}
+		for (int i = 1; i < dateHeader.length; i++) {
+			Cell cell = uHeadRow.createCell(i);
+			cell.setCellValue(dateHeader[i] + "-" + year);
+			cell.setCellStyle(headCellStyle);
+		}
+		for (int row = 1; row < duration.size(); row++) {
+			Cell cell = client.createCell(row);
+			cell.setCellType(Cell.CELL_TYPE_STRING);
+			cell.setCellValue(projects.get(row));
+			cell.setCellStyle(headCellStyle);
+		}
+		
+
+		try {
+			FileOutputStream out = new FileOutputStream(OUTPUT_FILE);
+			wb.write(out);
+			wb.close();
+		} catch(FileNotFoundException e) {
+			e.printStackTrace();
+		} catch(IOException e) {
+			e.printStackTrace();
+		}
```

Comment:
It looks like we're performing the same operation `printStackTrace()` on two different exception types. Do both exceptions share a common base class that provides the `printStackTrace()` method?

If we catch a general `Exception`, we can reduce two catches into one:

```java
try {
  // Operation that throws...
} catch (Exception e) {
  e.printStackTrace();
}
```

My Java knowledge is limited, so let me know your thoughts on this one: I noticed this function was labeled as `throw IOException` in the signature. But, it looks like we're catching and printing the stack trace without a re-throw. Is it necessary to still have the `throw IOException` in the signature?

Reply:
Discussed in person. This implementation is effectively a parser, and there isn't much better way of filtering through the weak schema expected by the input file